client-lite/src/util/http_agent.cpp

```diff
@@ -59,40 +62,61 @@ bool HttpAgent::ValidateUrl(const std::string& url)
 }
 
 // IHttpAgent
-HRESULT HttpAgent::SendRequest(PCSTR szUrl, PCSTR szProxyUrl, PCSTR szPostData, PCSTR szRange, UINT64 callerContext) try
+HRESULT HttpAgent::SendRequest(PCSTR szUrl, PCSTR szProxyUrl, PCSTR szRange, UINT64 callerContext) try
 {
     RETURN_IF_FAILED(_CreateClient(szUrl, szProxyUrl));
-    (void)_SubmitRequestTask(_CreateRequest(szPostData, szRange), callerContext);
+    DO_ASSERT(_requestContext.curlHandle);
+    if (szRange == nullptr)
+    {
+        curl_slist_free_all(_requestContext.requestHeaders);
+        curl_easy_setopt(_requestContext.curlHandle, CURLOPT_HTTPHEADER, nullptr);
+    }
+    else
+    {
+        std::string rangeHeader("Range: bytes=");
+        rangeHeader += szRange;
+        auto tempList = curl_slist_append(_requestContext.requestHeaders, rangeHeader.c_str());
+        RETURN_HR_IF(E_OUTOFMEMORY, tempList == nullptr);
+        _requestContext.requestHeaders = tempList;
+        curl_easy_setopt(_requestContext.curlHandle, CURLOPT_HTTPHEADER, _requestContext.requestHeaders);
+    }
+
+    _requestContext.responseHeaders.clear();
+    _requestContext.responseStatusCode = 0;
+    _requestContext.hrTranslatedStatusCode = S_OK;
+    _requestContext.responseOnHeadersAvailableInvoked = false;
+    _requestContext.responseOnCompleteInvoked = false;
+    _callbackContext = callerContext;
```

Comment:
>callerContext; [](http://example.com/codeflow?start=23&length=14)

Do we need to expose this as a function param? Looks like it's never set and default value is always used. It looks like a param for the IHttpEvents interface but unused by the functions implementing that interface

Reply:
Will remove in the next PR.